### DISCUSSION TITLE
Custom Field

### DISCUSSION THREAD

**Discussion by learingmmex** [moneymanagerex/moneymanagerex#3916] [Q&A] [answered]
Hi, is there anyway to filter transactions by the content in the custom field. Thanks

**Comment by TomAff** [comment]
Maybe With adding with a report 

When adding a variable &textfield
like
"&begin_date"
"&single_date"
"&end_date"
"&only_years"

**Reply by learingmmex** [reply]
Thank you Tom for prompt reply, i am week in developing customized reports, i would appreciate if you share with me a simple code to use it in the report to filter transactions with custom fields (two customs fields would be perfect.

**Comment by TomAff** [comment, marked as answer]
I'm not an expert. Try this script in the custom reports

`with alldata as (select
       CANS.TransID as ID,
       date(CANS.TransDate, 'localtime') as Date,
       d.userdate as UserDate
       ,coalesce(CAT.CategName, SCAT.CategName) as Category,
       coalesce(SUBCAT.SUBCategName, SSCAT.SUBCategName, '') as Subcategory,
       ROUND((case CANS.TRANSCODE when 'Withdrawal' then -1 else 1 end)*(case CANS.CATEGID when -1 then st.splittransamount else CANS.TRANSAMOUNT end),2) as Amount,
       CANS.Status as Status, CANS.NOTES as Notes,
       ifnull(CAT.CategId,st.CategId) as CategID,
       ifnull (ifnull(SUBCAT.SubCategID,st.subCategId),-1) as SubCategID,
       ifnull (PAYEE.PayeeName,'') as Payee,
            d.year as Year,
            d.month as Month,
            (select content from CUSTOMFIELDDATA_V1 cfd where refid=CANS.Transid and FIELDID=1) as customfield1,
            (select content from CUSTOMFIELDDATA_V1 cfd where refid=CANS.Transid and FIELDID=2) as customfield2
            from  CHECKINGACCOUNT_V1 CANS
            left join CATEGORY_V1 CAT on CAT.CATEGID = CANS.CATEGID
            left join SUBCATEGORY_V1 SUBCAT on SUBCAT.SUBCATEGID = CANS.SUBCATEGID and SUBCAT.CATEGID = CANS.CATEGID
            left join PAYEE_V1 PAYEE on PAYEE.PAYEEID = CANS.PAYEEID
            left join ACCOUNTLIST_V1 FROMACC on FROMACC.ACCOUNTID = CANS.ACCOUNTID
            left join splittransactions_V1 st on CANS.transid=st.transid
            left join CATEGORY_V1 SCAT on SCAT.CATEGID = st.CATEGID and CANS.TransId=st.transid
            left join SUBCATEGORY_V1 SSCAT on SSCAT.SUBCATEGID = st.SUBCATEGID and SSCAT.CATEGID = st.CATEGID and CANS.TransId=st.transid
            left join (select transid as id , date(transdate, 'localtime') as transdate
,round (strftime('%m', transdate, 'localtime'))  as month
,round (strftime('%Y', transdate, 'localtime'))  as year
,round(strftime('%Y', transdate, 'localtime' ,'start of month', ((case when fd.infovalue<=round(strftime('%d', transdate , 'localtime')) then 1  else 0 end)-fm.infovalue)||' month')) as finyear
,ifnull (ifnull (strftime(df.infovalue, TransDate, 'localtime'),
       (strftime(replace (df.infovalue, '%y', SubStr (strftime('%Y',TransDate, 'localtime'),3,2)),TransDate, 'localtime'))
       ), date(TransDate, 'localtime')) as UserDate
from  CHECKINGACCOUNT_V1
left join infotable_V1 df on df.infoname='DATEFORMAT'
left join infotable_V1 fm on fm.infoname='FINANCIAL_YEAR_START_MONTH'
left join infotable_V1 fd on fd.infoname='FINANCIAL_YEAR_START_DAY') d on d.id=CANS.TRANSID
order by date desc, CANS.transid
)

select * from alldata where customfield1<>'' or customfield2<>''`

**Reply by learingmmex** [reply]
Thank you very much, i tried it and it worked
